contents/tutorials/svelte-surveys.md

```diff
@@ -0,0 +1,486 @@
+    {/each}
+  </div>
+  <div>
+    <button class="button" on:click={emitDismiss}>Dismiss</button>
+    <button class="button" on:click={emitSubmit}>Submit</button>
+  </div>
+</div>
+
+<style>
+  .survey {
+    position: fixed;
+    bottom: 20px;
+    right: 20px;
+    width: 400px;
+    padding: 20px;
+    background-color: #ffffff;
+    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.2);
+    border-radius: 5px;
+    z-index: 1000;
+  }
+
+  .button {
+    margin: 5px;
+    padding: 5px;
+  }
+</style>
+```
+
+Then, integrate the component into `src/routes/+page.svelte`:
+
+```svelte file=+page.svelte
+<script>
+  import CustomSurvey from '../components/CustomSurvey.svelte';
+
+  let showSurvey = true;
+  let surveyTitle = 'Survey title';
+
+  const handleDismiss = () => {
+    showSurvey = false;
+  };
+
+  const handleSubmit = (event) => {
+    console.log("Submitted value:", event.detail.selectedValue);
+    showSurvey = false;
+  };
+</script>
+
+<div>
+  <h1>Svelte survey</h1>
+  {#if showSurvey}
+    <CustomSurvey
+      title={surveyTitle}
+      on:dismiss={handleDismiss}
+      on:submit={handleSubmit}
+    />
+  {/if}
+</div>
+</script>
+```
+
+This shows a survey popup every time you visit your app's homepage.
+
+![Custom survey UI](../images/tutorials/svelte-surveys/sample-survey-ui.png)
+
+#### 2. Fetch the survey from PostHog
+
+To fetch the active surveys, we use [`posthog.getActiveMatchingSurveys()`](/docs/libraries/js#surveys). This returns an array of survey objects that looks like this:
+
+```JSON
+[
+   {
+     "id": "018cfcd5-107e-0000-49a1-8e7c6b825947",
+     "name": "Net promoter score (NPS) API Survey",
+     "description": "",
+     "type": "api",
+     "linked_flag_key": null,
+     "targeting_flag_key": null,
+     "questions": [
+       {
+         "type": "rating",
+         "scale": 10,
+         "display": "number",
+         "question": "How likely are you to recommend us to a friend?",
+         "description": "",
+         "lowerBoundLabel": "Unlikely",
+         "upperBoundLabel": "Very likely"
+       }
+     ],
+     "conditions": null,
+     "start_date": "2024-01-12T08:41:20.614000Z",
+     "end_date": null
+   }
+]
+```
+
+To fetch this array and integrate it with your survey UI, we call `posthog.getActiveMatchingSurveys()` when our page is mounted:
+
+```svelte file=+page.svelte
+<script>
+  import CustomSurvey from '../components/CustomSurvey.svelte';
+  import { onMount } from 'svelte';
+  import posthog from 'posthog-js';
+
+  let showSurvey = true;
+  let surveyTitle = 'Survey title';
+  let surveyID = '';
+
+  onMount(() => {
+    fetchActiveSurveys();
+  });
+
+  const fetchActiveSurveys = async () => {
+    posthog.getActiveMatchingSurveys((surveys) => {
+      if (surveys.length > 0) {
+        const survey = surveys[0];
+        surveyID = survey.id;
+        surveyTitle = survey.questions[0].question;
+        showSurvey = true;
+      }
+    });
+  };
+
+  // rest of your code
+</script>
```

Comment:
```suggestion
```

Reply:
We need to include this otherwise it breaks the code formatting 😅